deepmind/encoder.go

```diff
@@ -173,13 +166,18 @@ func encodeBlock(bh types.EventDataNewBlock) ([]byte, error) {
 }
 
 func encodeTx(result *abci.TxResult) ([]byte, error) {
-	tx := &codec.EventTx{
-		TxResult: &codec.TxResult{
+	mappedTx, err := mapTx(result.Tx)
+	if err != nil {
+		return nil, err
+	}
+
+	tx := &pbcodec.EventTx{
+		TxResult: &pbcodec.TxResult{
 			Hash:   tmhash.Sum(result.Tx),
 			Height: uint64(result.Height),
 			Index:  result.Index,
-			Tx:     result.Tx,
-			Result: &codec.ResponseDeliverTx{
+			Tx:     mappedTx,
```

Comment:
Actually we could add this an additional field (e.g. `DecodedTx`). That way we could still parse an original `TxResult` on the Graph Node side and we'll have the encoded transaction payload in case we need it in a subgraph.

Reply:
I see this as just duplicated data and can't find a use case for encoded payload. Correct me if I'm wrong but the bulk of the data is Tx data right ? If we keep both doesn't that increase storage / bandwith for no real benefit ?

Reply:
That's right. I'm wondering if encoded transactions would be of any use to subgraph developers. If not, there's no clear benefit.